src/eve/utils.py

```diff
@@ -281,3 +401,680 @@ def reset_sequence(cls, start: int = 1) -> None:
+        """Filter elements with callables (equivalent to ``filter(func, self)``).
+
+        For detailed information check :func:`filter` reference.
+
+        Examples:
+            >>> it = xiter(range(4))
+            >>> list(it.filter(lambda x: x % 2 == 0))
+            [0, 2]
+
+            >>> it = xiter(range(4))
+            >>> list(it.filter(lambda x: x % 2 == 0).filter(lambda x: x > 1))
+            [2]
+
+
+        Notes:
+            `lambdas`, `partial` and `curried` functions are supported (see :meth:`map`).
+
+        """
+        if not callable(func):
+            raise ValueError(f"Invalid function or callable: '{func}'.")
+        return XIterator(filter(func, self.iterator))
+
+    def filter_by_type(self, *types: Type) -> XIterator[T]:
+        """Filter elements using :func"`isinstance` checks (equivalent to ``filter(isinstancechecker([*types]), self)``).
+
+        Examples:
+            >>> it = xiter([1, '2', 3.3, [4, 5], {6, 7}])
+            >>> list(it.filter_by_type(int, float))
+            [1, 3.3]
+
+        """
+        return XIterator(filter(isinstancechecker([*types]), self.iterator))
+
+    def getattr(  # noqa  # A003: shadowing a python builtin
+        self, *names: str, default: Any = NOTHING
+    ) -> XIterator[Any]:
+        """Get provided attributes from each item in a sequence (equivalent to ``map(sattrgetter(*names, default=default), self)``).
+
+        For detailed information check :func:`sattrgetter` and :func:`operator.attrgetter` reference.
+
+        Examples:
+            >>> from collections import namedtuple
+            >>> Point = namedtuple('Point', ['x', 'y'])
+            >>> it = xiter([Point(1.0, -1.0), Point(2.0, -2.0), Point(3.0, -3.0)])
+            >>> list(it.getattr('y'))
+            [-1.0, -2.0, -3.0]
+
+            >>> it = xiter([Point(1.0, -1.0), Point(2.0, -2.0), Point(3.0, -3.0)])
+            >>> list(it.getattr('x', 'z', default=None))
+            [(1.0, None), (2.0, None), (3.0, None)]
+
+        """
+        return XIterator(map(sattrgetter(*names, default=default), self.iterator))
+
+    def getitem(self, *indices: Union[int, str], default: Any = NOTHING) -> XIterator[Any]:
+        """Get provided indices data from each item in a sequence (equivalent to ``toolz.itertoolz.pluck(indices, self)``).
+
+        For detailed information check :func:`toolz.itertoolz.pluck` reference.
+
+            >>> it = xiter([('a', 1), ('b', 2), ('c', 3)])
+            >>> list(it.getitem(0))
+            ['a', 'b', 'c']
+
+            >>> it = xiter([
+            ...     dict(name="AA", age=20, country="US"),
+            ...     dict(name="BB", age=30, country="UK"),
+            ...     dict(name="CC", age=40, country="EU"),
+            ...     dict(country="CH")
+            ... ])
+            >>> list(it.getitem("name", "age", default=None))
+            [('AA', 20), ('BB', 30), ('CC', 40), (None, None)]
+
+        """
+        ind: Union[Any, List[Any]]
```

Comment:
Is this type hint useful? Isn't `Union[Any,X] == Any`?

Reply:
You're right, this is not useful at all for type checkers, it was meant to inform users that it will be most likely a hashable item or a list of hashable items. Changed to a source comment